### ISSUE TITLE
"mark" rule's autocorrections are eating preceding closing braces

### ISSUE THREAD

**Issue by ChristopherRogers** [realm/SwiftLint#1029]
_EDIT: This bug should now be reproducible with the code snippet below as of 0.16.1._

With code like this...

```swift
//MARK:- Top-Level bad mark
//MARK:- Another bad mark
struct MarkTest {
}
// MARK:- Bad mark
extension MarkTest {
}
```

`swiftlint autocorrect` will transform it into this, eating the closing braces. 😋 

```swift
// MARK: - Top-Level bad mark// MARK: - Another bad mark
struct MarkTest {
// MARK: - Bad mark
extension MarkTest {
}
```

**Comment by marcelofabri** [comment]
I couldn't reproduce it with 0.14 😬 
Are you using the latest version?

**Comment by jpsim** [comment]
@ChristopherRogers I'm just following up on @marcelofabri's request for a repro case. If there's a bug here we'd love to know about it!

**Comment by ChristopherRogers** [comment]
(Sorry for taking so long to get back; I had been on vacation.)

Sorry, I should've seen if the sample code reproduced the issue first. So, I took a look at the code that was triggering this bug and was able to reduce it to something smaller, which I've placed above. The number of mark comments correlates to the number of characters that are eaten.

The original code doesn't have multiple mark comments on the same type like the code above. Instead it has mark comments placed above methods.

**Comment by marcelofabri** [comment]
Thanks for the follow up, @ChristopherRogers. I could reproduce it now.

**Comment by kimdv** [comment]
Can still reproduce this in 0.20.1.
